web/secondary_toolbar.js

```diff
@@ -29,173 +29,232 @@
-      { element: this.toggleButton, handler: this.toggle },
-      // All items within the secondary toolbar
-      { element: this.presentationModeButton,
-        handler: this.presentationModeClick },
-      { element: this.openFile, handler: this.openFileClick },
-      { element: this.print, handler: this.printClick },
-      { element: this.download, handler: this.downloadClick },
-      { element: this.viewBookmark, handler: this.viewBookmarkClick },
-      { element: this.firstPage, handler: this.firstPageClick },
-      { element: this.lastPage, handler: this.lastPageClick },
-      { element: this.pageRotateCw, handler: this.pageRotateCwClick },
-      { element: this.pageRotateCcw, handler: this.pageRotateCcwClick },
-      { element: this.toggleHandTool, handler: this.toggleHandToolClick },
-      { element: this.documentPropertiesButton,
-        handler: this.documentPropertiesClick }
-    ];
-
-    for (var item in elements) {
-      var element = elements[item].element;
-      if (element) {
-        element.addEventListener('click', elements[item].handler.bind(this));
+    this.eventBus = eventBus;
+
+    this.opened = false;
+    this.previousContainerHeight = null;
+    this.newContainerHeight = null;
+    this.buttonContainer = this.toolbar.firstElementChild;
+
+    // Bind the event listeners for click and hand tool actions.
+    this.bindClickListeners();
+    this.bindHandToolListener();
+  }
+
+  SecondaryToolbar.prototype = {
+    /**
+     * @return {boolean}
+     */
+    get isOpen() {
+      return this.opened;
+    },
+
+    bindClickListeners: function SecondaryToolbar_bindClickListeners() {
+      var elements = [
+        // Button to toggle the visibility of the secondary toolbar.
+        { element: this.toggleButton, handler: this.toggle },
+        // All items within the secondary toolbar.
+        { element: this.presentationModeButton,
+          handler: this.presentationModeButtonClick },
+        { element: this.openFileButton, handler: this.openFileButtonClick },
+        { element: this.printButton, handler: this.printButtonClick },
+        { element: this.downloadButton, handler: this.downloadButtonClick },
+        { element: this.viewBookmarkButton,
+          handler: this.viewBookmarkButtonClick },
+        { element: this.firstPageButton, handler: this.firstPageButtonClick },
+        { element: this.lastPageButton, handler: this.lastPageButtonClick },
+        { element: this.pageRotateCwButton,
+          handler: this.pageRotateCwButtonClick },
+        { element: this.pageRotateCcwButton,
+          handler: this.pageRotateCcwButtonClick },
+        { element: this.toggleHandToolButton,
+          handler: this.toggleHandToolButtonClick },
+        { element: this.documentPropertiesButton,
+          handler: this.documentPropertiesButtonClick }
+      ];
+
+      for (var item in elements) {
+        var element = elements[item].element;
+        if (element) {
```

Comment:
I think that you can actually remove this `if` now. If memory serves, it was added in some sort of attempt to prevent the code from breaking entirely if a custom implementation decided to remove DOM elements. But we've never really supported that practice in general, so I don't think we need it here either.

Especially with the recent addition of the viewerConfig, see https://github.com/mozilla/pdf.js/blob/c1c199d70277a2a42fe1ac12ef598248076ac829/web/viewer.js#L79, it should be fairly easy to remove DOM elements: Just replace the ones you don't wish to use (in that list) with a hidden dummy element instead.